src/pygambit/tests/test_behav.py

```diff
@@ -1,703 +1,701 @@
-        assert self.profile_rational.action_value("U2") == gbt.Rational("3/1")
-        assert self.profile_rational.action_value("D2") == gbt.Rational("3/1")
-        assert self.profile_rational.action_value("U3") == gbt.Rational("7/2")
-        assert self.profile_rational.action_value("D3") == gbt.Rational("3/1")
-
-    def test_regret(self):
-        for profile in [self.profile_double, self.profile_rational]:
-            for player in self.game.players:
-                for infoset in player.infosets:
-                    for action in infoset.actions:
-                        assert (
-                            profile.regret(action) ==
-                            max(profile.action_value(a) for a in infoset.actions) -
-                            profile.action_value(action)
-                        )
-
-    def test_node_value(self):
-        # Another good node_value test (to be written!) is its martingale property: it should
-        # be the expected value of its children's node_values, given the probability
-        # distribution at the node.
-        for profile in [self.profile_double, self.profile_rational]:
-            for player in self.game.players:
+# tolerance for floating point assertions
+TOL = 1e-13
+
+
+def _create_mixed_behav_game():
+    """
+    Returns
+    -------
+    Game
+        Three-player extensive form game: binary tree with 3 infomation sets, one per player,
+        with 1, 2, and 4 nodes respectively
+    """
+    return gbt.Game.read_game("test_games/mixed_behavior_game.efg")
+
+
+def _create_complicated_exten_game():
+    """
+    Returns
+    -------
+    Game
+        Two-player extensive poker game with a chance move with two moves, then player 1 can raise
+        or fold; after raising player 2 is in an infoset with two nodes and can choose to meet or
+        pass
+    """
+    return gbt.Game.read_game("test_games/complicated_extensive_game.efg")
+
+
+def _set_action_probs(profile: gbt.MixedBehaviorProfile, probs: list, rational_flag: bool):
+    """Set the action probabilities in a behavior profile called ```profile``` according to a
+    list with probabilities in the order of ```profile.game.actions```
+    """
+    for i, p in enumerate(probs):
+        # assumes rationals given as strings
+        profile[profile.game.actions[i]] = gbt.Rational(p) if rational_flag else p
+
+
+@pytest.mark.parametrize(
+    "game,player_idx,payoff,rational_flag",
+    [(_create_mixed_behav_game(), 0, 3.0, False),
+     (_create_mixed_behav_game(), 1, 3.0, False),
+     (_create_mixed_behav_game(), 2, 3.25, False),
+     (_create_mixed_behav_game(), 0, "3", True),
+     (_create_mixed_behav_game(), 1, "3", True),
+     (_create_mixed_behav_game(), 2, "13/4", True)
+     ]
+)
+def test_payoff(game: gbt.Game, player_idx: int, payoff: typing.Union[str, float],
+                rational_flag: bool):
+    """Test to ensure that payoffs are returned correctly"""
```

Comment:
I've been getting rid of trivial docstrings for the test functions where they basically just repeat the name of the test - also because in some cases the original GSoC student docstrings mostly just show they didn't have much background in game theory.